lib/searchkick/query.rb

```diff
@@ -805,70 +805,86 @@ def set_aggregations(payload, filters, post_filters)
 
       aggs = aggs.to_h { |f| [f, {}] } if aggs.is_a?(Array) # convert to more advanced syntax
       aggs.each do |field, agg_options|
-        size = agg_options[:limit] ? agg_options[:limit] : 1_000
-        shared_agg_options = agg_options.except(:limit, :field, :ranges, :date_ranges, :where)
+        payload[:aggs][field] = build_aggregation(field, agg_options)
```

Comment:
I extracted lines 808 - 842 into the `build_aggregation` method.

Other than that, the `set_aggregations` method remains unchanged. Filers are applied only to the outer aggregation.